src/handlers/pull-reviewer.ts

```diff
@@ -0,0 +1,283 @@
+    }
+  }
+
+  /**
+   * Check if a review can be performed on the pull request
+   * @returns boolean indicating if review can be performed
+   */
+  async canPerformReview(): Promise<boolean> {
+    const { logger, payload } = this.context;
+    const { number, repository, action } = payload;
+    const { owner, name } = repository;
+
+    logger.info(`${repository.owner.login}/${repository.name}#${number} - ${action}`);
+
+    // Use octokit.paginate to automatically handle pagination
+    const timeline = await this.context.octokit.paginate(this.context.octokit.rest.issues.listEvents, {
+      owner: owner.login,
+      repo: name,
+      issue_number: number,
+      per_page: 100, // Optional: customize items per page
+    });
+
+    const reviews = timeline.filter((event) => event.event === "reviewed");
+    const botReviews = reviews.filter((review) => review.actor.type === "Bot");
+
+    if (!botReviews.length) {
+      logger.info("No bot reviews found");
+      return true;
+    }
+
+    const lastReview = botReviews[botReviews.length - 1];
+    const lastReviewDate = new Date(lastReview.created_at);
+    const now = new Date();
+    const diff = now.getTime() - lastReviewDate.getTime();
+
+    if (diff < this._oneDay) {
+      throw this.context.logger.error("Only one review per day is allowed");
+    }
+
+    logger.info("One review per day check passed");
+    return true;
+  }
+
+  /**
+   * Convert a pull request to draft mode
+   * @param shouldConvert - Whether to convert the PR to draft
+   * @param params - Parameters including nodeId and octokit instance
+   */
+  async convertPullToDraft(nodeId: string, octokit: Context["octokit"]) {
+    const toDraft = /* GraphQL */ `mutation {
+      convertPullRequestToDraft(input: {pullRequestId: "${nodeId}"}) {
+        pullRequest {
+          id
+          number
+          isDraft
+          title
+        }
+      }
+    }`;
+
+    try {
+      await octokit.graphql(toDraft);
+      this.context.logger.info(`Successfully converted pull request to draft mode.`);
+    } catch (e) {
+      throw this.context.logger.error("Failed to convert pull request to draft mode: ", { e });
+    }
+  }
+
+  /**
+   * Review a pull request using AI completion
+   * @returns The completion result from the AI model
+   */
+  async reviewPull() {
+    const {
+      env: { UBIQUITY_OS_APP_NAME },
+      config: { anthropicAiModel },
+      adapters: {
+        anthropic: { completions },
+      },
+    } = this.context;
+
+    const taskNumber = await this.getTaskNumberFromPullRequest(this.context);
+    const issue = await fetchIssue(this.context, taskNumber);
+
+    if (!issue) {
+      throw this.context.logger.error(`Error fetching issue, Aborting`, {
+        owner: this.context.payload.repository.owner.login,
+        repo: this.context.payload.repository.name,
+        issue_number: taskNumber,
+      });
+    }
+
+    const taskSpecification = issue.body ?? "";
```

Comment:
A task spec should never be empty so this logic is wrong. 

Empty strings are basically banned in our repos and there should be a CI check to enforce this.

Reply:
there is a case where an empty string is used
```${conditions ? `Conditions:\n${conditions.join("\n")}` : ""}``` 
```perFileDiffs.push({ filename, diffContent: "" });```